{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Functions"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "On this lab we will put to practice some of the concepts we have learned on this past few days.\n",
    "\n",
    "`NOTE: On this lab you should try to write all the functions yourself using only the most basic of python syntax and without functions such as len, count, sum, max, min, in, etc. Give it a try. üßëüèª‚Äçüíªüë©üèª‚Äçüíª`\n",
    "\n",
    "The cell after each exercise contains a few tests to check if your function works as expected."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "from mod.testing import *\n",
    "import unittest"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1. Write a function that returns the greater of two numbers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "def greater(a,b):\n",
    "    \n",
    "    if a > b:\n",
    "        return a\n",
    "    else:\n",
    "        return b\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "....................................................................................................\n",
      "----------------------------------------------------------------------\n",
      "Ran 100 tests in 0.041s\n",
      "\n",
      "OK\n"
     ]
    }
   ],
   "source": [
    "# This will test your function \n",
    "test_greater(greater)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2. Now write a function that returns the largest element on a list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "def greatest(arr):\n",
    "    x = arr[0]\n",
    "    for i in arr[1:]:\n",
    "        if i > x:\n",
    "            x = i\n",
    "        \n",
    "    return x\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "....................................................................................................\n",
      "----------------------------------------------------------------------\n",
      "Ran 100 tests in 0.052s\n",
      "\n",
      "OK\n"
     ]
    }
   ],
   "source": [
    "# This will test your function \n",
    "test_greatest(greatest)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 3. Write a function that sums all the elements on a list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "def sum_all(arr):\n",
    "    suma = 0\n",
    "    for i in arr:\n",
    "        suma += i\n",
    "    return suma"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "....................................................................................................\n",
      "----------------------------------------------------------------------\n",
      "Ran 100 tests in 0.041s\n",
      "\n",
      "OK\n"
     ]
    }
   ],
   "source": [
    "# This will test your function \n",
    "test_sum(sum_all)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 4. Write another function that multiplies all the elements on a list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "def mult_all(arr):\n",
    "    multiply = 1\n",
    "    for i in arr:\n",
    "        multiply *= i\n",
    "    return multiply"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "....................................................................................................\n",
      "----------------------------------------------------------------------\n",
      "Ran 100 tests in 0.045s\n",
      "\n",
      "OK\n"
     ]
    }
   ],
   "source": [
    "# This will test your function \n",
    "test_mult(mult_all)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 5. Now combine those two ideas and write a function that receives a list and either \"+\" or \"*\" and outputs acordingly"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [],
   "source": [
    "def oper_all(arr, oper):\n",
    "    suma = 0\n",
    "    multiply = 1\n",
    "    result = 0\n",
    "    for i in arr:\n",
    "        if oper == \"+\":\n",
    "            suma += i\n",
    "            result = suma\n",
    "        elif oper == \"*\":\n",
    "            multiply *= i\n",
    "            result = multiply\n",
    "    return result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "....................................................................................................\n",
      "----------------------------------------------------------------------\n",
      "Ran 100 tests in 0.043s\n",
      "\n",
      "OK\n"
     ]
    }
   ],
   "source": [
    "# This will test your function \n",
    "test_operations(oper_all)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 6. Write a function that returns the factorial of a number."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "6"
      ]
     },
     "execution_count": 96,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def factorial(n):\n",
    "    #asigno una variable factor = 1 que me sirve de base\n",
    "    #uso 1 porque n*1 es n\n",
    "    factor = 1\n",
    "    \n",
    "    while n > 1:\n",
    "        factor*=n\n",
    "        n-=1\n",
    "    return factor\n",
    "\n",
    "factorial(3)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "....................................................................................................\n",
      "----------------------------------------------------------------------\n",
      "Ran 100 tests in 0.037s\n",
      "\n",
      "OK\n"
     ]
    }
   ],
   "source": [
    "# This will test your function \n",
    "test_factorial(factorial)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 7. Write a function that takes a list and returns a list of the unique values.\n",
    "\n",
    "`NOTE: You cannot use set. ü§î`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "def unique(arr):\n",
    "    list_unique = []\n",
    "    for i in arr:\n",
    "        if i not in list_unique:\n",
    "            list_unique.append(i)\n",
    "    return list_unique"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "....................................................................................................\n",
      "----------------------------------------------------------------------\n",
      "Ran 100 tests in 0.106s\n",
      "\n",
      "OK\n"
     ]
    }
   ],
   "source": [
    "# This will test your function \n",
    "test_unique(unique)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 8. Write a function that returns the mode of a list, i.e.: the element that appears the most times.\n",
    "`NOTE: You should not use count... üßê`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [],
   "source": [
    "errey = [1,2,3,4,3,2,1,3,3,5,6,6,7]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [],
   "source": [
    "def resuelve(errey):\n",
    "    contador = 0\n",
    "    for i in errey:\n",
    "        \n",
    "        contador += 1\n",
    "        print(i, contador)\n",
    "        #print(i)\n",
    "    return i"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 1\n",
      "2 2\n",
      "3 3\n",
      "4 4\n",
      "3 5\n",
      "2 6\n",
      "1 7\n",
      "3 8\n",
      "3 9\n",
      "5 10\n",
      "6 11\n",
      "6 12\n",
      "7 13\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "7"
      ]
     },
     "execution_count": 91,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "resuelve(errey)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [],
   "source": [
    "def mode_counter(arr):\n",
    "    #contador = 0\n",
    "    count=0\n",
    "    for i in arr:\n",
    "        #print(i)\n",
    "        if i == i:\n",
    "            count += 1\n",
    "            #print(i, count)\n",
    "            \n",
    "    return i\n",
    "    #return i\n",
    "        \n",
    "        #contador += 1\n",
    "        \n",
    "        #print(i)\n",
    "    \n",
    "            \n",
    "        \n",
    "            \n",
    "        \n",
    "        \n",
    "        \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "....................................................................................................\n",
      "----------------------------------------------------------------------\n",
      "Ran 100 tests in 0.037s\n",
      "\n",
      "OK\n"
     ]
    }
   ],
   "source": [
    "# This will test your function \n",
    "test_mode(mode_counter)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 9. Write a function that calculates the standard deviation of a list.\n",
    "`NOTE: Do not use any libraries or already built functions. üòâ`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [],
   "source": [
    "#from statistics import stdev\n",
    "def st_dev(arr):\n",
    "    counter = 0\n",
    "    sum_counter = 0\n",
    "    \n",
    "    for i in arr:\n",
    "        sum_counter += i\n",
    "        counter += 1\n",
    "    media = sum_counter / counter\n",
    "    suma_cuadrados = 0\n",
    "    \n",
    "    for i in arr:\n",
    "        suma_cuadrados += (i - media) ** 2\n",
    "    variance = (suma_cuadrados/counter)\n",
    "    std = variance ** (1/2)\n",
    "    return std\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      ".FF...F..F..FF.F.FFFF...FFFFFFFFF.FFF..FF...FFFF.FF.FFFF.FFFF..FF.FFF...FFF.FF..F..F..F.....FF..F..F\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_stdev.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 142, in runTest\n",
      "    self.assertAlmostEqual(fn(self.input), self.output, delta=5, msg=f\"Should be {self.output}\")\n",
      "AssertionError: 588.9650867839395 != 595.7741579252877 within 5 delta (6.809071141348227 difference) : Should be 595.7741579252877\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_stdev.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 142, in runTest\n",
      "    self.assertAlmostEqual(fn(self.input), self.output, delta=5, msg=f\"Should be {self.output}\")\n",
      "AssertionError: 533.0024330150849 != 538.4137639627072 within 5 delta (5.411330947622332 difference) : Should be 538.4137639627072\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_stdev.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 142, in runTest\n",
      "    self.assertAlmostEqual(fn(self.input), self.output, delta=5, msg=f\"Should be {self.output}\")\n",
      "AssertionError: 614.5322408141009 != 634.6861691681436 within 5 delta (20.153928354042705 difference) : Should be 634.6861691681436\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_stdev.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 142, in runTest\n",
      "    self.assertAlmostEqual(fn(self.input), self.output, delta=5, msg=f\"Should be {self.output}\")\n",
      "AssertionError: 618.648065196545 != 642.0014378129118 within 5 delta (23.35337261636687 difference) : Should be 642.0014378129118\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_stdev.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 142, in runTest\n",
      "    self.assertAlmostEqual(fn(self.input), self.output, delta=5, msg=f\"Should be {self.output}\")\n",
      "AssertionError: 513.8044352985154 != 526.4928751835562 within 5 delta (12.688439885040793 difference) : Should be 526.4928751835562\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_stdev.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 142, in runTest\n",
      "    self.assertAlmostEqual(fn(self.input), self.output, delta=5, msg=f\"Should be {self.output}\")\n",
      "AssertionError: 526.9885302550788 != 534.6822004351602 within 5 delta (7.693670180081426 difference) : Should be 534.6822004351602\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_stdev.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 142, in runTest\n",
      "    self.assertAlmostEqual(fn(self.input), self.output, delta=5, msg=f\"Should be {self.output}\")\n",
      "AssertionError: 574.9448140962734 != 588.4747785037732 within 5 delta (13.529964407499847 difference) : Should be 588.4747785037732\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_stdev.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 142, in runTest\n",
      "    self.assertAlmostEqual(fn(self.input), self.output, delta=5, msg=f\"Should be {self.output}\")\n",
      "AssertionError: 548.3617879821962 != 556.608292647397 within 5 delta (8.246504665200746 difference) : Should be 556.608292647397\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_stdev.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 142, in runTest\n",
      "    self.assertAlmostEqual(fn(self.input), self.output, delta=5, msg=f\"Should be {self.output}\")\n",
      "AssertionError: 573.6194195809694 != 578.8106513617992 within 5 delta (5.191231780829753 difference) : Should be 578.8106513617992\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_stdev.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 142, in runTest\n",
      "    self.assertAlmostEqual(fn(self.input), self.output, delta=5, msg=f\"Should be {self.output}\")\n",
      "AssertionError: 618.7137812070745 != 632.020976307554 within 5 delta (13.307195100479476 difference) : Should be 632.020976307554\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_stdev.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 142, in runTest\n",
      "    self.assertAlmostEqual(fn(self.input), self.output, delta=5, msg=f\"Should be {self.output}\")\n",
      "AssertionError: 505.82928391188454 != 517.7327780208913 within 5 delta (11.903494109006715 difference) : Should be 517.7327780208913\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_stdev.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 142, in runTest\n",
      "    self.assertAlmostEqual(fn(self.input), self.output, delta=5, msg=f\"Should be {self.output}\")\n",
      "AssertionError: 536.5008644913818 != 547.5639019024782 within 5 delta (11.063037411096388 difference) : Should be 547.5639019024782\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_stdev.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 142, in runTest\n",
      "    self.assertAlmostEqual(fn(self.input), self.output, delta=5, msg=f\"Should be {self.output}\")\n",
      "AssertionError: 562.9801349958984 != 568.6958158984539 within 5 delta (5.715680902555505 difference) : Should be 568.6958158984539\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_stdev.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 142, in runTest\n",
      "    self.assertAlmostEqual(fn(self.input), self.output, delta=5, msg=f\"Should be {self.output}\")\n",
      "AssertionError: 588.9244356704053 != 603.4679697185182 within 5 delta (14.543534048112974 difference) : Should be 603.4679697185182\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_stdev.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 142, in runTest\n",
      "    self.assertAlmostEqual(fn(self.input), self.output, delta=5, msg=f\"Should be {self.output}\")\n",
      "AssertionError: 590.7020929774407 != 596.4651766981663 within 5 delta (5.763083720725604 difference) : Should be 596.4651766981663\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_stdev.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 142, in runTest\n",
      "    self.assertAlmostEqual(fn(self.input), self.output, delta=5, msg=f\"Should be {self.output}\")\n",
      "AssertionError: 572.938128217552 != 578.3179469796369 within 5 delta (5.379818762084824 difference) : Should be 578.3179469796369\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_stdev.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 142, in runTest\n",
      "    self.assertAlmostEqual(fn(self.input), self.output, delta=5, msg=f\"Should be {self.output}\")\n",
      "AssertionError: 546.3670160548116 != 552.1488386135295 within 5 delta (5.781822558717863 difference) : Should be 552.1488386135295\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_stdev.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 142, in runTest\n",
      "    self.assertAlmostEqual(fn(self.input), self.output, delta=5, msg=f\"Should be {self.output}\")\n",
      "AssertionError: 634.3106914222328 != 640.0511766774855 within 5 delta (5.740485255252679 difference) : Should be 640.0511766774855\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_stdev.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 142, in runTest\n",
      "    self.assertAlmostEqual(fn(self.input), self.output, delta=5, msg=f\"Should be {self.output}\")\n",
      "AssertionError: 494.31479878094643 != 501.97905115953824 within 5 delta (7.6642523785918115 difference) : Should be 501.97905115953824\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_stdev.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 142, in runTest\n",
      "    self.assertAlmostEqual(fn(self.input), self.output, delta=5, msg=f\"Should be {self.output}\")\n",
      "AssertionError: 572.1596246093023 != 577.4330966828808 within 5 delta (5.273472073578432 difference) : Should be 577.4330966828808\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_stdev.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 142, in runTest\n",
      "    self.assertAlmostEqual(fn(self.input), self.output, delta=5, msg=f\"Should be {self.output}\")\n",
      "AssertionError: 589.3082329453406 != 595.0577177095778 within 5 delta (5.749484764237195 difference) : Should be 595.0577177095778\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_stdev.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 142, in runTest\n",
      "    self.assertAlmostEqual(fn(self.input), self.output, delta=5, msg=f\"Should be {self.output}\")\n",
      "AssertionError: 444.17430137278313 != 468.20082348401644 within 5 delta (24.026522111233305 difference) : Should be 468.20082348401644\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_stdev.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 142, in runTest\n",
      "    self.assertAlmostEqual(fn(self.input), self.output, delta=5, msg=f\"Should be {self.output}\")\n",
      "AssertionError: 597.9308404824089 != 630.2744463944082 within 5 delta (32.34360591199925 difference) : Should be 630.2744463944082\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_stdev.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 142, in runTest\n",
      "    self.assertAlmostEqual(fn(self.input), self.output, delta=5, msg=f\"Should be {self.output}\")\n",
      "AssertionError: 567.8677705099582 != 575.9230348436256 within 5 delta (8.055264333667424 difference) : Should be 575.9230348436256\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_stdev.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 142, in runTest\n",
      "    self.assertAlmostEqual(fn(self.input), self.output, delta=5, msg=f\"Should be {self.output}\")\n",
      "AssertionError: 602.0032484793568 != 607.6559851563188 within 5 delta (5.652736676962036 difference) : Should be 607.6559851563188\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_stdev.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 142, in runTest\n",
      "    self.assertAlmostEqual(fn(self.input), self.output, delta=5, msg=f\"Should be {self.output}\")\n",
      "AssertionError: 626.0887853107072 != 637.5776112109492 within 5 delta (11.488825900241977 difference) : Should be 637.5776112109492\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_stdev.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 142, in runTest\n",
      "    self.assertAlmostEqual(fn(self.input), self.output, delta=5, msg=f\"Should be {self.output}\")\n",
      "AssertionError: 595.1644147034503 != 608.5405660704631 within 5 delta (13.376151367012767 difference) : Should be 608.5405660704631\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_stdev.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 142, in runTest\n",
      "    self.assertAlmostEqual(fn(self.input), self.output, delta=5, msg=f\"Should be {self.output}\")\n",
      "AssertionError: 467.42448897395394 != 490.2389398872943 within 5 delta (22.81445091334035 difference) : Should be 490.2389398872943\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_stdev.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 142, in runTest\n",
      "    self.assertAlmostEqual(fn(self.input), self.output, delta=5, msg=f\"Should be {self.output}\")\n",
      "AssertionError: 514.0410126698175 != 519.9838807674024 within 5 delta (5.942868097584892 difference) : Should be 519.9838807674024\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_stdev.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 142, in runTest\n",
      "    self.assertAlmostEqual(fn(self.input), self.output, delta=5, msg=f\"Should be {self.output}\")\n",
      "AssertionError: 498.87040401290596 != 503.9352080154263 within 5 delta (5.064804002520361 difference) : Should be 503.9352080154263\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_stdev.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 142, in runTest\n",
      "    self.assertAlmostEqual(fn(self.input), self.output, delta=5, msg=f\"Should be {self.output}\")\n",
      "AssertionError: 555.6604424860114 != 564.551574117644 within 5 delta (8.891131631632561 difference) : Should be 564.551574117644\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_stdev.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 142, in runTest\n",
      "    self.assertAlmostEqual(fn(self.input), self.output, delta=5, msg=f\"Should be {self.output}\")\n",
      "AssertionError: 572.1383546617025 != 581.29314962973 within 5 delta (9.154794968027545 difference) : Should be 581.29314962973\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_stdev.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 142, in runTest\n",
      "    self.assertAlmostEqual(fn(self.input), self.output, delta=5, msg=f\"Should be {self.output}\")\n",
      "AssertionError: 556.0286528820153 != 566.620626586197 within 5 delta (10.591973704181669 difference) : Should be 566.620626586197\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_stdev.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 142, in runTest\n",
      "    self.assertAlmostEqual(fn(self.input), self.output, delta=5, msg=f\"Should be {self.output}\")\n",
      "AssertionError: 604.9442414954948 != 613.064661774236 within 5 delta (8.120420278741221 difference) : Should be 613.064661774236\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_stdev.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 142, in runTest\n",
      "    self.assertAlmostEqual(fn(self.input), self.output, delta=5, msg=f\"Should be {self.output}\")\n",
      "AssertionError: 528.856577911252 != 539.7619845820934 within 5 delta (10.905406670841444 difference) : Should be 539.7619845820934\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_stdev.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 142, in runTest\n",
      "    self.assertAlmostEqual(fn(self.input), self.output, delta=5, msg=f\"Should be {self.output}\")\n",
      "AssertionError: 470.56635900248125 != 489.7809950140196 within 5 delta (19.214636011538346 difference) : Should be 489.7809950140196\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_stdev.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 142, in runTest\n",
      "    self.assertAlmostEqual(fn(self.input), self.output, delta=5, msg=f\"Should be {self.output}\")\n",
      "AssertionError: 556.4816866425433 != 561.8069915825198 within 5 delta (5.3253049399764905 difference) : Should be 561.8069915825198\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_stdev.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 142, in runTest\n",
      "    self.assertAlmostEqual(fn(self.input), self.output, delta=5, msg=f\"Should be {self.output}\")\n",
      "AssertionError: 574.5241783930294 != 582.0346171376395 within 5 delta (7.5104387446101555 difference) : Should be 582.0346171376395\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_stdev.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 142, in runTest\n",
      "    self.assertAlmostEqual(fn(self.input), self.output, delta=5, msg=f\"Should be {self.output}\")\n",
      "AssertionError: 590.88921662739 != 613.1947192664527 within 5 delta (22.305502639062638 difference) : Should be 613.1947192664527\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_stdev.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 142, in runTest\n",
      "    self.assertAlmostEqual(fn(self.input), self.output, delta=5, msg=f\"Should be {self.output}\")\n",
      "AssertionError: 550.7939660885772 != 567.7454250340189 within 5 delta (16.951458945441686 difference) : Should be 567.7454250340189\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_stdev.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 142, in runTest\n",
      "    self.assertAlmostEqual(fn(self.input), self.output, delta=5, msg=f\"Should be {self.output}\")\n",
      "AssertionError: 509.8161394990845 != 526.5358447753898 within 5 delta (16.71970527630532 difference) : Should be 526.5358447753898\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_stdev.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 142, in runTest\n",
      "    self.assertAlmostEqual(fn(self.input), self.output, delta=5, msg=f\"Should be {self.output}\")\n",
      "AssertionError: 602.4820303866128 != 609.7851095188704 within 5 delta (7.303079132257608 difference) : Should be 609.7851095188704\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_stdev.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 142, in runTest\n",
      "    self.assertAlmostEqual(fn(self.input), self.output, delta=5, msg=f\"Should be {self.output}\")\n",
      "AssertionError: 619.2770102563946 != 637.2307864215277 within 5 delta (17.953776165133036 difference) : Should be 637.2307864215277\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_stdev.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 142, in runTest\n",
      "    self.assertAlmostEqual(fn(self.input), self.output, delta=5, msg=f\"Should be {self.output}\")\n",
      "AssertionError: 560.9583680251499 != 578.2226507354493 within 5 delta (17.264282710299426 difference) : Should be 578.2226507354493\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_stdev.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 142, in runTest\n",
      "    self.assertAlmostEqual(fn(self.input), self.output, delta=5, msg=f\"Should be {self.output}\")\n",
      "AssertionError: 544.203800098517 != 549.9627308455991 within 5 delta (5.758930747082104 difference) : Should be 549.9627308455991\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_stdev.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 142, in runTest\n",
      "    self.assertAlmostEqual(fn(self.input), self.output, delta=5, msg=f\"Should be {self.output}\")\n",
      "AssertionError: 562.276847978192 != 572.9878456172271 within 5 delta (10.7109976390351 difference) : Should be 572.9878456172271\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_stdev.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 142, in runTest\n",
      "    self.assertAlmostEqual(fn(self.input), self.output, delta=5, msg=f\"Should be {self.output}\")\n",
      "AssertionError: 638.490583086456 != 645.8722223395678 within 5 delta (7.381639253111757 difference) : Should be 645.8722223395678\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_stdev.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 142, in runTest\n",
      "    self.assertAlmostEqual(fn(self.input), self.output, delta=5, msg=f\"Should be {self.output}\")\n",
      "AssertionError: 520.4037951437327 != 533.9230281604268 within 5 delta (13.519233016694102 difference) : Should be 533.9230281604268\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_stdev.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 142, in runTest\n",
      "    self.assertAlmostEqual(fn(self.input), self.output, delta=5, msg=f\"Should be {self.output}\")\n",
      "AssertionError: 574.324715045941 != 581.1216938252521 within 5 delta (6.796978779311075 difference) : Should be 581.1216938252521\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_stdev.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 142, in runTest\n",
      "    self.assertAlmostEqual(fn(self.input), self.output, delta=5, msg=f\"Should be {self.output}\")\n",
      "AssertionError: 635.642259401881 != 642.5142673270212 within 5 delta (6.872007925140224 difference) : Should be 642.5142673270212\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_stdev.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 142, in runTest\n",
      "    self.assertAlmostEqual(fn(self.input), self.output, delta=5, msg=f\"Should be {self.output}\")\n",
      "AssertionError: 558.9986584887604 != 564.5609711945012 within 5 delta (5.562312705740851 difference) : Should be 564.5609711945012\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_stdev.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 142, in runTest\n",
      "    self.assertAlmostEqual(fn(self.input), self.output, delta=5, msg=f\"Should be {self.output}\")\n",
      "AssertionError: 572.2620477360006 != 579.3711656278517 within 5 delta (7.109117891851156 difference) : Should be 579.3711656278517\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_stdev.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 142, in runTest\n",
      "    self.assertAlmostEqual(fn(self.input), self.output, delta=5, msg=f\"Should be {self.output}\")\n",
      "AssertionError: 556.568123762841 != 564.463101413319 within 5 delta (7.89497765047804 difference) : Should be 564.463101413319\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_stdev.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 142, in runTest\n",
      "    self.assertAlmostEqual(fn(self.input), self.output, delta=5, msg=f\"Should be {self.output}\")\n",
      "AssertionError: 491.7485588807458 != 499.8771849164553 within 5 delta (8.128626035709487 difference) : Should be 499.8771849164553\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_stdev.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 142, in runTest\n",
      "    self.assertAlmostEqual(fn(self.input), self.output, delta=5, msg=f\"Should be {self.output}\")\n",
      "AssertionError: 507.51027653526097 != 514.5107489749955 within 5 delta (7.000472439734551 difference) : Should be 514.5107489749955\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_stdev.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 142, in runTest\n",
      "    self.assertAlmostEqual(fn(self.input), self.output, delta=5, msg=f\"Should be {self.output}\")\n",
      "AssertionError: 422.6946176566699 != 438.6509350920648 within 5 delta (15.956317435394908 difference) : Should be 438.6509350920648\n",
      "\n",
      "----------------------------------------------------------------------\n",
      "Ran 100 tests in 0.047s\n",
      "\n",
      "FAILED (failures=56)\n"
     ]
    }
   ],
   "source": [
    "# This will test your function \n",
    "test_stdev(st_dev)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 10. Write a function to check if a string is a pangram, i.e.: if it contains all the letters of the alphabet at least once. Mind that the strings may contain characters that are not letters."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'\\npor que si pongo esta funcion no pasa los test? No deberian ser iguales?\\n\\ndef pangram(string):\\n    alphabet = \"abcdefghijklmnopqrstuvwxyz\"\\n    \\n    for i in alphabet:\\n        if i in string.lower():\\n            return True\\n        \\n            \\n    return False\\n'"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def pangram(string):\n",
    "    abecedario = \"abcdefghijklmnopqrstuvwxyz\"\n",
    "    \n",
    "    for i in abecedario:\n",
    "        if i not in string.lower():\n",
    "            return False\n",
    "    return True\n",
    "'''\n",
    "por que si pongo esta funcion no pasa los test? No deberian ser iguales?\n",
    "\n",
    "def pangram(string):\n",
    "    alphabet = \"abcdefghijklmnopqrstuvwxyz\"\n",
    "    \n",
    "    for i in alphabet:\n",
    "        if i in string.lower():\n",
    "            return True\n",
    "        \n",
    "            \n",
    "    return False\n",
    "'''\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "..............................\n",
      "----------------------------------------------------------------------\n",
      "Ran 30 tests in 0.019s\n",
      "\n",
      "OK\n"
     ]
    }
   ],
   "source": [
    "# This will test your function \n",
    "test_pangram(pangram)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 11. Write a function that receives a string of comma separated words and returns a string of comma separated words sorted alphabetically.\n",
    "\n",
    "`NOTE: You may use sorted but not split and definitely no join! ü§™`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {},
   "outputs": [],
   "source": [
    "def sort_alpha(string):\n",
    "    pass"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'bbtpz,ckrw,dkhp,fjagb,ghapff,irstqhhs,jd[74 chars]zuhz' : Should be bbtpz,ckrw,dkhp,fjagb,ghapff,irstqhhs,jdejwpvsj,kztg,lseeyzzh,nrpodndyy,oxweefk,ptun,qrqn,uawrs,xcklsqm,ygpnsujov,zuhz\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'clku,dyutlfywh,ffxhygc,fmti,foolaraf,hmi[67 chars]mvog' : Should be clku,dyutlfywh,ffxhygc,fmti,foolaraf,hmiqiyy,isepky,jabng,kmlfiv,lgptihepot,nbrup,otqsaps,rlzfbs,shukdso,wimvog\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'bqupezrrb,cjsef,dfhsrjemz,igdqkyhqmz,izt[59 chars]nyob' : Should be bqupezrrb,cjsef,dfhsrjemz,igdqkyhqmz,iztq,kueyciz,lomeowex,ndkvxd,owtytr,rtlqz,tgrrxfu,ueyvjh,wctavnyob\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'aggiowh,bnklndda,bovvrzbdwm,fqsmfngsi,js[93 chars]xzke' : Should be aggiowh,bnklndda,bovvrzbdwm,fqsmfngsi,jstr,lqwhxjv,lrnumlx,nkizf,psxzxutxw,qbrqc,ueerlpsii,uqmh,wevrdlvj,wjnquhnm,wplzvcgvqq,xfhmhco,xzke\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'bwyrid,cgbqhsjl,dctt,fcwpzndvw,hrmt,hvty[132 chars]azdc' : Should be bwyrid,cgbqhsjl,dctt,fcwpzndvw,hrmt,hvtyj,jwkyvnotea,lrcleuli,mgaagzitzq,mgbmh,oyaiqecry,peyjq,pgxm,pygu,qmyfntesbr,rptyhv,spymxpevp,vhaniru,wxbsc,ybrenrziw,yhnyp,ylmhlz,yuazdc\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'bezuuzpsil,dhqxbbwd,donqdn,hbdvnkacc,icd[57 chars]brra' : Should be bezuuzpsil,dhqxbbwd,donqdn,hbdvnkacc,icdqs,juyy,ksgqgbsu,ltjxvyc,ncwhn,ovenlx,tvhrp,waccepyogy,ycbrra\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'buqsh,duyumrcmfj,icjihyufl,iemlspepgd,lr[49 chars]gdwe' : Should be buqsh,duyumrcmfj,icjihyufl,iemlspepgd,lrllxn,myxbumgm,nnzbmo,pvrkdgut,udaez,usehbcgv,uybggdwe\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'airh,biwkrfkybz,ctrong,emgmaq,hkkuru,icd[83 chars]ylxq' : Should be airh,biwkrfkybz,ctrong,emgmaq,hkkuru,icdwcscegf,kfwxtn,pdblsp,pelntrnq,ppul,qmaccmkdu,skacdk,urmvmyjszp,vbwcsdiiq,ykiwpavf,ylxq\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'arvl,atjhmq,bjvmqtf,eccseao,fwlgrpuemu,g[116 chars]ypkt' : Should be arvl,atjhmq,bjvmqtf,eccseao,fwlgrpuemu,gkffb,kbozdhduck,kwrgeiqaer,lhcihor,mxuh,nctwaxcwr,ococdozl,pyrcxw,qliuhjzsq,tegeadp,vjsdxt,wokqsskw,xikca,ykvbnjpeq,ypkt\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'avfwnowue,czflgtiq,ewojw,fudq,fuysni,gec[141 chars]clhf' : Should be avfwnowue,czflgtiq,ewojw,fudq,fuysni,gecum,gqhxeiq,hnkn,iwetvl,jvetzqirl,lxokgw,msindvtpve,ojkdfpmvc,rbaqlr,rfyaprok,rjqxh,sosfaeqhar,tlxx,vwztttuqm,wkdcemn,wpcganrt,yfzu,ylvjzdi,yrclhf\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'gqrxb,kazxm,nkjytwwrt,qnrbtsmzp,rpmgdfi,stogkima,uemunuiij,xapcikrhc' : Should be gqrxb,kazxm,nkjytwwrt,qnrbtsmzp,rpmgdfi,stogkima,uemunuiij,xapcikrhc\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'aneewcv,kloqsvg,lrmfzgtbqp,mmaqkcqqjc,sm[52 chars]aeou' : Should be aneewcv,kloqsvg,lrmfzgtbqp,mmaqkcqqjc,smekcdygy,tlvbcrlun,tsoxqku,urwtmjs,withxbhwg,xrswm,ykaeou\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'abnqmg,arzwjxv,bgjiahne,bzklxk,cdniuat,d[146 chars]muco' : Should be abnqmg,arzwjxv,bgjiahne,bzklxk,cdniuat,dsqveh,gxukyobbbj,hshu,imrzj,jyexcv,kmgerj,kphjgiurf,npzpb,oezpavrlr,omgodqwd,sqcwwwmn,tganmjjuns,tqjvjc,vxzc,xrguznvb,xxaea,ymvybjki,yvoeoyc,yzthhmuco\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'bayqmcvch,dbdrq,dczoentzz,fcodshuasa,fswfgb,nlkglf,qrutr,qtmo,zfmuwbvlwl' : Should be bayqmcvch,dbdrq,dczoentzz,fcodshuasa,fswfgb,nlkglf,qrutr,qtmo,zfmuwbvlwl\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'ehvsoaue,hohpw,hsoquvxuk,hufkmxpq,kqddxz[63 chars]ttex' : Should be ehvsoaue,hohpw,hsoquvxuk,hufkmxpq,kqddxzmeso,kvrzx,lbwaafowzc,nxmzgtwk,qnvmoaan,rtizwbuepg,ulasle,ypgxettex\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'amqn,arlfapwsqy,czfqgypot,dgwhdjjqpe,edd[113 chars]wckf' : Should be amqn,arlfapwsqy,czfqgypot,dgwhdjjqpe,eddkfzgyi,iecqvpbg,jfwifl,ktwtbnfte,lvguo,pontec,szbfawcx,tcnngmcv,tdgjxkqweg,xenxvmkw,ydkyzjmzz,znwlkjn,zudiuxw,zvrwckf\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'alvcu,bqjm,eochggecy,fujwye,gacnmysxgu,g[133 chars]odus' : Should be alvcu,bqjm,eochggecy,fujwye,gacnmysxgu,gnlry,hhgijdfbec,jhlxnrwofq,kyprxzmi,mubrlbrws,nomt,paxudpp,pvcypx,redsqkweq,sfkhmnar,sfydziw,tanquzh,ujeliu,uncsmdrt,webvxwws,wrwpk,yodus\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'atgkh,bcjvmx,cjwvm,efjfxaxf,efxsc,exspe,[118 chars]iepr' : Should be atgkh,bcjvmx,cjwvm,efjfxaxf,efxsc,exspe,jsbq,kaajtsz,kretmhsh,msonr,pxty,qqyxne,rtiy,sdxkbpxb,sykcp,tsskose,veawzibqbg,vixfyplzie,vnqjroma,xgrekvx,ymkuetny,yxiepr\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'acwdf,aifuurieqx,brsyo,easxttn,ehfrmwajy[125 chars]npyy' : Should be acwdf,aifuurieqx,brsyo,easxttn,ehfrmwajy,gdrwjok,hndbjjyxe,hsolo,hzyfnzu,jrzo,kgxzvtrs,lsgps,mzjq,nplpogb,rfjnpquid,rroz,rrykmi,thrfi,tkfxx,yjeqthdgje,zefflqejy,zsbrnpyy\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'fipk,hvfoi,ifpkdoxb,iwzzzcri,lmsqqraa,lr[96 chars]oscj' : Should be fipk,hvfoi,ifpkdoxb,iwzzzcri,lmsqqraa,lrouiwfuln,ppjsjtpupd,qlqnq,sfnzekx,shntjj,tevbudvju,ttupjqrg,ujmtrpmunp,vheyt,vjqkrisk,yhefp,ysaqoscj\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'aywqqkacs,dbpxhc,djuu,dqpwvbycn,ebqdyep,[106 chars]qnco' : Should be aywqqkacs,dbpxhc,djuu,dqpwvbycn,ebqdyep,fmtbf,fsfdca,iljgbbbrj,jritmo,lawp,mwlll,neagaer,oyzk,qmcfhaljg,qqiiogk,sjih,vhbhdumlvc,vrysye,wdopfbajr,yqnco\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'aulf,dzvn,kegg,lzjse,sifk,ssbbds,tnxd,xded' : Should be aulf,dzvn,kegg,lzjse,sifk,ssbbds,tnxd,xded\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'adyvxfnzlc,chivlyts,fltbnda,fvzwiokcq,ha[119 chars]hyaz' : Should be adyvxfnzlc,chivlyts,fltbnda,fvzwiokcq,hazqoto,iffx,jyww,knixy,krauktifc,kugosuf,kvyisjbdd,pdzexcb,qhtargz,rvmsoma,rwdrwpiw,szvctiblde,uaqkzah,wpgry,yjqbnuxd,zvhyaz\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'ankdm,edzntxj,gqjtkrn,gvnend,ihdepxwla,olpwd,upefcwkw,vkmqdhxefr' : Should be ankdm,edzntxj,gqjtkrn,gvnend,ihdepxwla,olpwd,upefcwkw,vkmqdhxefr\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'ahdo,bcruyw,dkssjhtb,etajn,ewqsico,fpqwl[70 chars]szeh' : Should be ahdo,bcruyw,dkssjhtb,etajn,ewqsico,fpqwl,gxpep,ihbwipoat,iwlmdj,meecsinqxg,pfkdxqbq,qjsxw,sjizw,vvgg,yvot,zmxaszeh\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'byjg,cvejqu,dogmmrrrd,fcjlpsd,ibjgnin,la[54 chars]uncr' : Should be byjg,cvejqu,dogmmrrrd,fcjlpsd,ibjgnin,laen,ogfbumnnt,oxxdbcogl,pavq,qedpwer,qpctvf,rgne,rnhuzkuncr\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'gtivtc,hvlzd,rueii,zahdy' : Should be gtivtc,hvlzd,rueii,zahdy\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'anpvb,henprytc,jbcb,kytsa,qqpf,qxfoksn,t[38 chars]zllo' : Should be anpvb,henprytc,jbcb,kytsa,qqpf,qxfoksn,tzbwlc,vqjgfiqg,warvftfion,wcphmma,zycyzllo\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'ashs,dfqcwcgrjx,dolse,gbclu,gwkkorpi,hps[164 chars]ilhs' : Should be ashs,dfqcwcgrjx,dolse,gbclu,gwkkorpi,hpsbuyijh,idqkzoh,igreofmcri,ihri,ijcvlf,lrulhpqas,lyodox,mksoohb,mkyhmn,mokhehbxx,ntizrnefrh,oxymfufudm,qiprrppj,qzwt,rvoimkzjh,vtleqzyx,wbatg,yabxbaqkme,zexfk,zvwhptilhs\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'cxwuz,divjgqr,dizehzkgld,edjuzckykn,heub[146 chars]mqlt' : Should be cxwuz,divjgqr,dizehzkgld,edjuzckykn,heubhvsdz,hjxarz,jagqeb,kmqq,ksrybxdtu,lcpqiyfsfd,lmztjj,ltnsplf,mdobulr,oqickqjjre,oumgmax,slxnvd,soyhvb,vprnnszmgk,vsrmrwfza,wkffc,wxav,xvpytozjm,zwmqlt\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'abipxtx,abnec,aiudygyy,atsnx,bppggwdq,br[137 chars]yofg' : Should be abipxtx,abnec,aiudygyy,atsnx,bppggwdq,brcqob,brurczdl,epni,eusid,ffhgkttpj,hglvedn,hrgggygs,hzbad,ienxfoju,lnfufs,nqpd,nxbsqrlv,oewchcv,oioz,osnc,viciprt,vslfgg,vwkzky,wjpcrsyf,yofg\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'amakh,amitvyyya,cmvjpzddzt,ehwi,hdnapzvr[127 chars]vsww' : Should be amakh,amitvyyya,cmvjpzddzt,ehwi,hdnapzvrn,ibzcjrt,isgri,jhhuhln,jrnkfpl,ktrlbnih,mrjxqvvxg,msekjqip,mysjlclbkg,nlfqjc,ntuh,puicuzws,rbuxty,sbdpknde,vfxxkshvat,vlsuywe,vsww\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'agsmf,awspj,axjzdlrm,bvmxyqa,dkjbhp,egbl[133 chars]wlbc' : Should be agsmf,awspj,axjzdlrm,bvmxyqa,dkjbhp,egblkk,jgypoy,jpfckfe,kjbbszrxb,mwqnp,nlgh,nvotamla,onsllpbd,pojmh,qxkbhicxcq,rcefzspi,simki,sjmc,soirdh,ueffnseu,vhfsgg,vqipseyh,wceibl,wlbc\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'dvfwjlopol,exuruuu,flpuaa,iavoprii,jgqkv[108 chars]xykt' : Should be dvfwjlopol,exuruuu,flpuaa,iavoprii,jgqkvpsk,jjpt,kenaidkqi,knaanxogr,kwmxerg,mmhnfpmid,ojctuqq,rknea,rzqvbnxtc,sddnybkofb,sfowzkz,vatcmxg,xiqljouwp,xykt\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'ajcxpqexa,cmicovf,iepn,iotsok,jgjhairwmy[163 chars]pxiy' : Should be ajcxpqexa,cmicovf,iepn,iotsok,jgjhairwmy,kfzzjn,kpievcnxsn,nkevscsmt,nmpazdl,ntjarazrct,nzuokv,onilfoofhg,pfwegfidz,qhsn,qlkrhyu,rjpsiudpin,tgjaxna,tijfx,tpsad,trzqlxbbb,tyzyrklon,wbxkfsped,wvuuqkeb,zdlspxiy\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'awucna,bnsqilfeje,brcgqssjtl,dfgbr,glfn,[143 chars]mnmi' : Should be awucna,bnsqilfeje,brcgqssjtl,dfgbr,glfn,glqcwkfer,hdhi,ittkldbyns,mqcbwryd,pvcuy,qfbxvlfzso,qhsc,qlkhcjfpc,shqlva,sjqttgypza,ssnhohw,sxbyxvsks,xsytxo,xvufvjabp,yfonqcsrq,ykxkhcuj,yxrhmnmi\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'augkqcllvc,bcilg,bjpfbqtqay,dcdryowgk,eo[156 chars]xmfr' : Should be augkqcllvc,bcilg,bjpfbqtqay,dcdryowgk,eohnnhaj,ggstbcyse,iwzes,msvbo,nbjgj,ngwscn,npzmyyfpw,pabyfcsyro,qqiyunbhxv,rexelgycj,rtyjcttff,srcfnsd,uyjyzoob,vhkxs,vidonvrv,vxeahhn,zgmedgrj,zkoymae,zyhltxmfr\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'fgfqlojes,glvogynbsh,kebuo,oqtfr,rmtuvf,szqqoop' : Should be fgfqlojes,glvogynbsh,kebuo,oqtfr,rmtuvf,szqqoop\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'bpurpk,dgydjn,duqilyb,ewiya,femqo,kqrdlf[88 chars]mgjg' : Should be bpurpk,dgydjn,duqilyb,ewiya,femqo,kqrdlfj,lvov,niaauwcs,qfii,srussbhysh,tgohtecqc,tpfhiih,tscgxcb,tzrvhupkv,wfknfrsn,ybzjoldj,znmgjg\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'ciunhsde,cmylzjmv,etdkk,fghjotdeet,fzyfz[136 chars]rchy' : Should be ciunhsde,cmylzjmv,etdkk,fghjotdeet,fzyfz,ghrcofv,giarh,hqwxpjad,knumkn,kouwrl,ksxg,lltg,lsliapkxg,ncywflwic,pbikbwti,qmvrcs,rxjscwn,totlcztd,wcahwz,ynyisc,yylzb,zlcytbcjje,zztprchy\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'bkorkvna,bubcjur,ctxpynn,demlwzbc,dnmyc,[139 chars]cnyh' : Should be bkorkvna,bubcjur,ctxpynn,demlwzbc,dnmyc,fmzxpysi,haglah,nnpdap,osncoceuy,pakvjwvqg,ptpudlu,qaevvssv,qswhfdvy,svbgnly,szqu,ulsmlnqsj,uqyk,uzrfr,vkoivuzng,vmhf,xruqszqqw,yanz,zpxabpcnyh\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'adqbozjq,busgfcpd,fxcwmeqj,jvjqq,lcfxvpgq,pdqtiobdx' : Should be adqbozjq,busgfcpd,fxcwmeqj,jvjqq,lcfxvpgq,pdqtiobdx\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'bvfaiygjv,bxmll,cits,emcmkhzhox,etxlfqdf[123 chars]labl' : Should be bvfaiygjv,bxmll,cits,emcmkhzhox,etxlfqdfu,fdbnm,fhrk,huybxko,iyjnkgchhs,jsocif,kaecowvp,kgscixc,krhr,lgjzzzb,lridjvhpq,nwlfdnbg,ocuvhcz,olxbezv,pbfklelocm,vlxov,xflabl\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'afrvcvcsn,arjvkl,body,eozuaycux,ewmtjhxo[96 chars]kaok' : Should be afrvcvcsn,arjvkl,body,eozuaycux,ewmtjhxo,hysjween,iuqgnlob,iyfl,jlxmxd,nrpuzxoyc,qlnbvhwide,rltdpchc,sxyfs,tcncszn,tyxa,xwkbhehtg,zsenvikaok\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'amahaurz,cqdrc,dxzqqf,dyllhyub,enesjmv,g[102 chars]rzxm' : Should be amahaurz,cqdrc,dxzqqf,dyllhyub,enesjmv,gdfjolnz,hqri,ktgcjyb,kvjznv,lavfiz,ntymuaeiqs,pkdav,qwubejjghb,raqb,sgpnv,tmtfchgra,whhujf,ytbjaohcn,zrzxm\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'ahacuya,avefyao,ckctjroy,gxtp,hwzu,igmo,[44 chars]koes' : Should be ahacuya,avefyao,ckctjroy,gxtp,hwzu,igmo,khcklrdhw,ldlmxbjxs,myrn,nvrvjynsn,uhddd,ztckoes\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'cqaeqjq,ghflfkxd,gokqljz,mdmjvbpng,mymir[63 chars]ddgt' : Should be cqaeqjq,ghflfkxd,gokqljz,mdmjvbpng,mymiratp,oswxvs,rcxfe,rjuan,snpbhzixp,svrbswrk,tfqqjxy,tvnjzgs,vbzinddgt\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'bdobi,cocn,egovpk,exxevdol,geeofhzkz,jcm[128 chars]orsz' : Should be bdobi,cocn,egovpk,exxevdol,geeofhzkz,jcmh,jmwzldjw,knzfi,kuopradz,lrcwkqepeg,ncfzasab,nxvzecdne,rarklkaesj,tutvn,tvsaufz,xkcjo,xrhr,ybwkknesij,ygrgt,ywkmghdwa,yyhbr,zpcorsz\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'axontpn,ghxldktmql,gkcstph,hkbs,plljbd,wweq,xgfuuwrwak,ywcsotslg,zpcsxfcxnj' : Should be axontpn,ghxldktmql,gkcstph,hkbs,plljbd,wweq,xgfuuwrwak,ywcsotslg,zpcsxfcxnj\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'abhlmn,awicwqtk,bdhvucw,gfjvx,gyxoxrc,hm[57 chars]horl' : Should be abhlmn,awicwqtk,bdhvucw,gfjvx,gyxoxrc,hmyvrhh,iigccl,iwqfb,izfgde,nsxvgcu,vhzhzmwih,wholki,xpqfzohorl\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'cmtxmbxnze,ecziu,fmaynphclh,kafohxmfg,ke[62 chars]nrhy' : Should be cmtxmbxnze,ecziu,fmaynphclh,kafohxmfg,keyklch,kuqmo,lwwbefmur,orgkhh,pjcxvdyojf,qbnbkzs,sycppod,yqunnwnrhy\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'bvtt,djmd,eryitb,fezst,hxpmjvt,ilwzzrch,[110 chars]mcmo' : Should be bvtt,djmd,eryitb,fezst,hxpmjvt,ilwzzrch,jmceczck,jmifcelibx,jyqhnj,ktbtryi,ktgcjvv,lrbsoxex,odchbs,odxtqeadrq,rnhvnao,sruimolw,sstp,ugpcukgugs,xretq,ymcmo\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'ddsal,dwezip,ijggx,ivmhwqvr,nmswcpx,qems[65 chars]mvcc' : Should be ddsal,dwezip,ijggx,ivmhwqvr,nmswcpx,qemskl,qwie,qxodo,rbtqm,uxhldwekc,wsbcodl,xmtqh,ysecfn,zjaaywvy,zzcghmvcc\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'cuneqt,deyyjvsy,fshkq,lgmg,qbuohfmtxt,qgjzjei,qwmxxlia,ssdazciaw,vamiu' : Should be cuneqt,deyyjvsy,fshkq,lgmg,qbuohfmtxt,qgjzjei,qwmxxlia,ssdazciaw,vamiu\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'aukcnasn,cereys,cwdap,exokmqrcwd,gaydnu,[130 chars]voje' : Should be aukcnasn,cereys,cwdap,exokmqrcwd,gaydnu,gprbd,gsyys,imbyuqfln,lwxgoqsmbx,pprhbr,pprpeqlz,rampsd,rkzukmlpi,usnakeq,vdrug,veexd,wbgyl,wczqupdo,wtvu,xbyimlb,zanhcioa,zggjh,zvoje\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'ahuq,bgzwzu,bwtxp,dkahzt,dlaeoz,ecph,ftm[135 chars]iawb' : Should be ahuq,bgzwzu,bwtxp,dkahzt,dlaeoz,ecph,ftmelarfm,gniwabjq,gxyoulana,hijojcy,iaezicy,iuggx,mflfujz,motkgmxoe,nesaibm,nmttb,oufyuov,qricw,ruogzqi,uedrlf,wcjdm,wdmnjvjik,wquiefjy,xiawb\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'bevearkw,brjozfaxw,ipkipxlpaj,jhscvtey,k[114 chars]zojv' : Should be bevearkw,brjozfaxw,ipkipxlpaj,jhscvtey,kdlsqjiq,lnczyno,lqwzlpsvv,ngfjsmbhvw,otenazes,ovavldok,pssthrg,sbiofkl,spxy,szjdsgk,wbjfregwn,wptlumiz,ycrgb,zfcilzojv\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'afkzsjdhfr,bbryndfce,dnkxab,dqizbu,ezddx[98 chars]cwyg' : Should be afkzsjdhfr,bbryndfce,dnkxab,dqizbu,ezddxiv,hsxh,lzis,mitwkr,mjotpfhda,mpuob,ntwvtc,phmynl,rvxlzt,vsuowfhg,wenswsehc,wxwgvfe,xcxkdojcsy,ybgcwyg\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'brgl,dxiiprb,ejcpawz,fniir,goyc,hbziwhgi[98 chars]sasq' : Should be brgl,dxiiprb,ejcpawz,fniir,goyc,hbziwhgiea,kuxwnkjabn,kyzg,lgagcngo,lnobv,lrlhwqgwpc,nkvfa,ptxf,ruguezhv,soncn,vdvfwfe,vphg,yphijrbeq,ztjjsasq\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'aqpssot,jbfymbrgl,jpnkqes,ktxnn,lrescgng,outclrc,pqizocbim,ubdxei' : Should be aqpssot,jbfymbrgl,jpnkqes,ktxnn,lrescgng,outclrc,pqizocbim,ubdxei\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'asigrmwg,fupmvudfyf,gtwrj,hlxxipeaj,irzv[113 chars]tfwu' : Should be asigrmwg,fupmvudfyf,gtwrj,hlxxipeaj,irzvxsln,knob,lbstajqjzj,lkpfn,mhsawus,nxocgnup,ookaszahg,qatjdx,rfzyqoufz,stokigzdeu,trhpzxx,txst,ugygs,wyfvyob,zmqgtfwu\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'arunrgaocm,bgnc,czxbtzw,evdaknikm,gxppra[96 chars]ywoy' : Should be arunrgaocm,bgnc,czxbtzw,evdaknikm,gxppraswew,huciexz,jfrwcishvx,krgwjjludy,kseyw,nyidxi,qzsz,rdgpgkcr,twepjzi,vehi,vhoelkqdwd,yzjdiyk,zqywoy\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'cjocgcuwx,gbifhqmue,igiyyr,kpcafqqt,oowjxamf,yiceg,yrwrnl' : Should be cjocgcuwx,gbifhqmue,igiyyr,kpcafqqt,oowjxamf,yiceg,yrwrnl\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'gumcoq,gylg,iaafhxpx,ixjhpmn,jbqstfiv,kn[59 chars]gruy' : Should be gumcoq,gylg,iaafhxpx,ixjhpmn,jbqstfiv,knqqfmml,nwsmgvijaw,pauwbzelmq,peqfyrhpef,rqfcyqo,vkygu,xhyetgruy\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'qxvrd,ubqulwc,vpoxu,vusbk,wkxzkd' : Should be qxvrd,ubqulwc,vpoxu,vusbk,wkxzkd\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'cfvhg,egwlafo,fajiicgi,hgxjiufv,metiuj,pqrxopeo,ugxe,wlczs,xkwsjgg,xrmyfggfza' : Should be cfvhg,egwlafo,fajiicgi,hgxjiufv,metiuj,pqrxopeo,ugxe,wlczs,xkwsjgg,xrmyfggfza\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'bpdj,bsszvdkd,gfpmshlsp,pjqid,racdoa,rrvxegvgvl,seiqt,zcovuobvnc,zysthpqpai' : Should be bpdj,bsszvdkd,gfpmshlsp,pjqid,racdoa,rrvxegvgvl,seiqt,zcovuobvnc,zysthpqpai\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'anod,asiuycsaqn,byyt,czui,dgqrkj,dlyfjsr[112 chars]mofz' : Should be anod,asiuycsaqn,byyt,czui,dgqrkj,dlyfjsrwsv,gtes,idipbd,olshcdgpq,pbbwokq,pksl,prvrasrjdg,ptbpqi,qikfjo,rofnibuvm,scmrelvq,vqxhxv,vsqwaod,wfwqsfxu,zxtapmofz\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'eucfu,fgbsnatco,fvkmlody,fvpasgf,ismwsjj[55 chars]rinb' : Should be eucfu,fgbsnatco,fvkmlody,fvpasgf,ismwsjjk,lgfshkia,ltklsozmo,umkmo,vgkgyv,vlxto,wkjijyccj,ziurjrinb\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'cenrwvhvz,fyzidc,ivrw,kbjrit,mqycw,mrnlk[51 chars]fals' : Should be cenrwvhvz,fyzidc,ivrw,kbjrit,mqycw,mrnlkyjugf,pdyypen,qzhndkll,rkbd,wemauqawv,wuukkhdh,xyqnfals\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'kjxtsuto,kvhyuls,lzgd,qbvvumuqn,rqmfjo' : Should be kjxtsuto,kvhyuls,lzgd,qbvvumuqn,rqmfjo\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'axzdkx,bcloukgcn,ccdqvcgmlq,evhzjz,hmhbb[91 chars]muuh' : Should be axzdkx,bcloukgcn,ccdqvcgmlq,evhzjz,hmhbbkte,jrvcsn,jymp,kicep,mfbd,mgzmonhmcf,mmps,ntmw,oncocp,ongaodpe,pgkg,qkdvjtikg,qsau,uiaeu,ymuuh\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'awppd,byxx,cevqcpk,cwyaqri,dlak,dnsyajc,[99 chars]qeoj' : Should be awppd,byxx,cevqcpk,cwyaqri,dlak,dnsyajc,dnwpia,fklymvtl,gcjrlb,gojvt,ljvvnxapn,psmscecfbf,qavmxlrt,qnfgd,qzwl,seske,sudgijmct,wlnyhpdy,zzbaqeoj\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'adaskxc,crhcy,fsgmwuzp,ghplajzg,hzzgzc,j[154 chars]ijal' : Should be adaskxc,crhcy,fsgmwuzp,ghplajzg,hzzgzc,jhck,kvyauxdqj,lpglh,lzfixdgc,mibjseinr,nixfm,oiusksc,oxgchxgrso,rkogklc,rmgpua,sjisg,slpbqblok,ubbygu,wrhj,xjuhnpr,xtwiagmcg,yioihke,zcqyvxe,zhxxfdoju,zzyijal\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'bemdegwgne,bmkcuifpt,chtbuaytdd,hvjeqnfa[96 chars]modt' : Should be bemdegwgne,bmkcuifpt,chtbuaytdd,hvjeqnfa,mckzgftxk,nlkjtjw,pgwolfh,pkofhqqzo,qcksgkgy,qufvl,rfwrorfbpi,swdwuoeeiq,uxdzp,zgagxyxchh,ztrlomodt\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'brxoaf,diuci,dpcaa,eiypgfms,gixizjmdqq,h[116 chars]tamh' : Should be brxoaf,diuci,dpcaa,eiypgfms,gixizjmdqq,heee,iuazrrrzz,jkqza,kiwzo,mxhgrh,nxzpdg,oonjxhmyld,qisid,qphgobtp,qzrqcx,xelfvtpgwm,xjgrvgyiqp,xqlmxr,ycmwthzptd,ylytamh\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'fihj,pmfrsapjqc,psggc,xvoweeutn,yuurqvffh,zgwrdkp' : Should be fihj,pmfrsapjqc,psggc,xvoweeutn,yuurqvffh,zgwrdkp\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'dgjzbmn,gxapnx,jkwuql,kwrqo,lghqspqka,obbjj,oghpboi,uiqr,uqnolasc,zurzrhayhn' : Should be dgjzbmn,gxapnx,jkwuql,kwrqo,lghqspqka,obbjj,oghpboi,uiqr,uqnolasc,zurzrhayhn\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'gaqttxeuo,mdcuthr,ocyvh,saed,szcg,zajdyc' : Should be gaqttxeuo,mdcuthr,ocyvh,saed,szcg,zajdyc\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'acizvebpn,aegwt,cgfc,jkvxdteers,rnilzei,rsjluzou,sdqwptt,tjuhiflwb,uswwzyuwl' : Should be acizvebpn,aegwt,cgfc,jkvxdteers,rnilzei,rsjluzou,sdqwptt,tjuhiflwb,uswwzyuwl\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'aslfs,byuwdbqswx,gdyetzub,nufwxajjck,otlqnjhae,tgghjm,uaiheu' : Should be aslfs,byuwdbqswx,gdyetzub,nufwxajjck,otlqnjhae,tgghjm,uaiheu\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'acjuhsc,afcnvqss,afdffwv,bwcxby,ctgyn,fj[119 chars]xhsq' : Should be acjuhsc,afcnvqss,afdffwv,bwcxby,ctgyn,fjxlgi,fraltjfn,gbednn,iaxdv,ltbupqo,nijdref,oklnbufbp,renfnq,tbyvh,veaq,wpjzgvco,wpmjhjfplc,xyvgsqb,ybmnz,yzrs,zlwq,zwuexhsq\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'bmdjszucdr,cgavfml,djhbl,dujnldug,efjxh,[66 chars]ddum' : Should be bmdjszucdr,cgavfml,djhbl,dujnldug,efjxh,fpqa,gkxuy,gwjtj,ilfjiypmy,mzac,njtgtogfe,ojws,tlwlno,xdiksaiq,yitddum\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'dbqrsyli,elarjjy,eqbblnrn,geesrnsuc,geoy[119 chars]wnbm' : Should be dbqrsyli,elarjjy,eqbblnrn,geesrnsuc,geoykurlpi,grnkwdmxw,hxzo,kkbubbtolv,ktzmmna,nxhitx,ociglonkxy,ohlf,rweled,rzojht,rzweskkr,scrgfwmo,vtoalwgtlw,wmlaku,zkfutwnbm\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'drtwloysty,duxcuvg,gszwybmxbz,jwfgkfwc,qmetqm,yddl' : Should be drtwloysty,duxcuvg,gszwybmxbz,jwfgkfwc,qmetqm,yddl\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'blopptgl,cyhajtqoq,dasojf,dkgmmoa,fbqt,f[90 chars]mcag' : Should be blopptgl,cyhajtqoq,dasojf,dkgmmoa,fbqt,ffhdyq,fkvco,ghrgcvzgd,hqnot,ikqncbto,kziz,leycvkvr,mtpp,mxsod,nyky,qted,rlin,witzr,wyhkw,xmcag\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'dhcujho,gmmsmjqkr,iejilh,itswvq,jqkxjxq,[167 chars]vdte' : Should be dhcujho,gmmsmjqkr,iejilh,itswvq,jqkxjxq,kfsyicgipe,klbxhvwij,ktbeifioix,kwdiqwtpor,onnpkvu,pcrtwnob,puwovhaawb,rfiuejoask,smycaj,tbqvuibfh,ucavbdlxu,ucqrs,uxmrj,uzbgkqiarg,xknc,xlwnjolsj,xpoa,xwdcrkhffq,yytavdte\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'atonkj,awagd,bteuhviu,bzpzqbk,ckjcoiyh,c[144 chars]clfk' : Should be atonkj,awagd,bteuhviu,bzpzqbk,ckjcoiyh,cuej,cwnapjlkp,efxxcmlg,elqcmfnpxn,enxchyf,gkfqnev,hqdmdtyjyq,hvpszndupd,ivliupljwx,ivpjfp,jbiwnj,mqfxx,thotxt,votvlag,vqmaqxrvb,yamxlgphlu,ytcsvclfk\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'aaktyh,aapczwjaf,acznmersd,ayuyrinmvm,bi[167 chars]hody' : Should be aaktyh,aapczwjaf,acznmersd,ayuyrinmvm,bixdomjr,ckpbfbcx,dkimdmxg,ejsofthqc,eoye,ftpypzwi,fxsom,grimcduuqu,hsqhkz,mjwxjdp,mwsvus,nejirxyt,qhkkjeza,tocukaakpy,ulpwcnw,voavvsqt,wejhabn,wycasuwko,xnukjz,yexl,zdjhody\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'eyxxkzmfhs,frapt,ksipjppp,lhkyx,ljbjnzy,[46 chars]zclq' : Should be eyxxkzmfhs,frapt,ksipjppp,lhkyx,ljbjnzy,oimrg,opzm,pqzdvuv,tzsbhvab,wfmeungubf,xhymzl,zclq\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'aozrmunwx,axrv,qcqaegkuz,rwqawuma,srzfjk' : Should be aozrmunwx,axrv,qcqaegkuz,rwqawuma,srzfjk\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'aflaryk,afvozajihn,cmgeh,ffbvafm,hjpfxnd[166 chars]rxdw' : Should be aflaryk,afvozajihn,cmgeh,ffbvafm,hjpfxndmuc,ialvzfm,isop,jdbyhbciqo,jeqhvrwt,jwbtgp,lqhfasajhf,mmeomvf,nfwxhsfju,ochdyglk,pjcnodlzoy,prxuspz,rjlthbh,rkrv,rrraa,tammrncjjm,tmvrasonae,wbedl,xpvttd,ydnvrmm,ytsrxdw\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'ackrbl,blaaxdosd,camzuggo,mcxkhc,pcwyhkppf,sfjv,tyrxsnsdfc,xdlohz,znvvst' : Should be ackrbl,blaaxdosd,camzuggo,mcxkhc,pcwyhkppf,sfjv,tyrxsnsdfc,xdlohz,znvvst\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'hlou,jconvoevn,kfmqludqe,mpgivbv,mpljzfx[53 chars]yoqk' : Should be hlou,jconvoevn,kfmqludqe,mpgivbv,mpljzfx,nmijj,nuvwfbfnal,udhnaufof,ujmgwwu,uxwdqqzstf,wanwn,yoqk\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'bcwg,bygsfmfzn,ccrxutn,cpzvjkq,ehez,eiav[102 chars]ugwa' : Should be bcwg,bygsfmfzn,ccrxutn,cpzvjkq,ehez,eiavipd,fomvjjf,gefnflajj,gskrcsc,hobmv,jlvjdx,kvkdmwb,lquwwdynqn,mqaro,ozrykfkn,tmgartmsl,ugkb,ukqpeedc,uugwa\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'ahfcsat,apazztam,buse,caqppu,dcexljzwwo,[82 chars]tmlt' : Should be ahfcsat,apazztam,buse,caqppu,dcexljzwwo,dirimo,dyvzny,ebpxwqv,gpbdoyr,hxcrq,ksebkielm,ovnrrlcktc,qsdqm,rgwqyla,tkvcbtytfg,tmlt\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'aljmkzy,avkkso,bvqyf,csbvm,fbnlrjrejc,hd[97 chars]suov' : Should be aljmkzy,avkkso,bvqyf,csbvm,fbnlrjrejc,hdess,hneggw,hsns,htjpu,hzdnxys,mccnscbao,mklijzpwtq,puqteo,qpqlfkc,tayjyklnqw,tslnfrivcv,vhnihrm,zsuov\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'bidtj,club,jnmwryw,lwxhtmrr,ouzki,rftlvear' : Should be bidtj,club,jnmwryw,lwxhtmrr,ouzki,rftlvear\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'bogmyapsk,cfzdv,dito,ehgtqvqx,iribueb,iz[120 chars]gfcj' : Should be bogmyapsk,cfzdv,dito,ehgtqvqx,iribueb,izpgr,jryitf,kfhcjz,kpmgthkfx,mdxrrbt,msnfoxmfrv,nmcreh,parcm,pxrphldqe,qgfqoeg,vojjxxi,vwycfp,xaacpm,yvfcdlxyb,zatocuo,zugfcj\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/sergio/Escritorio/Labs-Datamad-January/lab-functions/mod/testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: None != 'bziwl,cdxxppr,dffnzt,dtdhnrv,eeouvbsq,fl[79 chars]zsjt' : Should be bziwl,cdxxppr,dffnzt,dtdhnrv,eeouvbsq,fldb,fqrlldvq,gykv,hzxntp,nswlxr,orxo,pggvmorv,qkaiazlkr,qrns,rjaavtg,tcse,vpngj,zsjt\n",
      "\n",
      "----------------------------------------------------------------------\n",
      "Ran 100 tests in 0.055s\n",
      "\n",
      "FAILED (failures=100)\n"
     ]
    }
   ],
   "source": [
    "# This will test your function \n",
    "test_alpha(sort_alpha)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 12. Write a function to check if a given password is strong (at least 8 characters, at least one lower case, at least one upper case, at least one number and at least one special character). It should output True if strong and False if not.\n",
    "`Valid special characters: # @ ! $ % & ( ) ^ * [ ] { }`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "ename": "SyntaxError",
     "evalue": "invalid syntax (<ipython-input-60-3c86cce769e5>, line 4)",
     "output_type": "error",
     "traceback": [
      "\u001b[0;36m  File \u001b[0;32m\"<ipython-input-60-3c86cce769e5>\"\u001b[0;36m, line \u001b[0;32m4\u001b[0m\n\u001b[0;31m    if any(i.isupper())\u001b[0m\n\u001b[0m                       ^\u001b[0m\n\u001b[0;31mSyntaxError\u001b[0m\u001b[0;31m:\u001b[0m invalid syntax\n"
     ]
    }
   ],
   "source": [
    "def check_pass(string):\n",
    "    pass"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'check_pass' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-62-daeb66aa08ec>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0;31m# This will test your function\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m \u001b[0mtest_pass\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcheck_pass\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m: name 'check_pass' is not defined"
     ]
    }
   ],
   "source": [
    "# This will test your function \n",
    "test_pass(check_pass)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
